### PR TITLE
Bugfix/cam ee2 fixes and plots runtime fix

### PR DESCRIPTION
**Pull request by MarcelCaron-NOAA**
## Pull Request Testing ##

- [x] Describe testing already performed for this Pull Request:</br>

**This bugfix PR addresses two separate issues:**
1. EE2 feedback from Lin Gan, various needed edits outlined in the Edits to EVS auth repo Google Doc (not linked here)
2. Excessive CAM grid2obs plots job runtime (>>15h) and a few killed job cards during runtime

**After this bugfix ...** 
- most EE2 feedback for the CAM det component should be satisfied by the changes here (excluding nam-firewxnest)
- the CAM grid2obs plots job runs more efficiently, with no killed jobs, and a runtime < 10 hours.

**Regarding prior testing:**
- Most jobs have been fully tested following the EE2 edits.
- The CAM grid2obs plots job has been tested extensively over the last two weeks during and following the runtime fix.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

This is an extensive testing PR, but I think all jobs need only be tested once for successful completion, and most complete rather quickly (plots take the longest).

### (1) Before testing:
- symlink the EVS_fix directory locally as "fix"
- **In all driver scripts,** point all exports of "`HOMEevs`" to your EVS test directory
- **In all precip stats driver scripts,** change cyc to "`19`"
- **In all precip stats driver scripts,** add "`export VDATE=20230926`" (use any VDATE earlier than PDYm2).  Alternatively, the CAM stats J-job can be edited to use PDYm3 for precip if that's easier.
- **In all other stats driver scripts,** change cyc to "`00`"
- **In all stats driver scripts,** change COMROOT to the parallel directory (`/lfs/h2/emc/vpppg/noscrub/emc.vpppg`)
- _Optional:_ **In all stats drivers scripts,** change "`maillist`" to point to your email address
- **In all plots driver scripts,** change COMIN to the parallel directory (`/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver/stats/$COMPONENT`)
- _Note: I believe all radar and severe jobs need to be fed the cyc variable_

### (2) From your EVS testing directory, the following scripts should be run using the above edits:
```
dev/drivers/scripts/cam/prep/jevs_cam_hireswarw_precip_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_hireswarw_severe_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_hireswarwmem2_precip_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_hireswarwmem2_severe_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_hireswfv3_precip_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_hireswfv3_severe_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_href_severe_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_hrrr_precip_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_hrrr_severe_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_namnest_precip_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_namnest_severe_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_radar_prep.sh
dev/drivers/scripts/cam/prep/jevs_cam_severe_prep.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarw_grid2obs_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarw_precip_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarw_radar_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarw_severe_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarw_snowfall_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarwmem2_grid2obs_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarwmem2_precip_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarwmem2_radar_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarwmem2_severe_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswarwmem2_snowfall_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswfv3_grid2obs_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswfv3_precip_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswfv3_radar_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswfv3_severe_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hireswfv3_snowfall_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hrrr_grid2obs_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hrrr_precip_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hrrr_radar_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hrrr_severe_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_hrrr_snowfall_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_namnest_grid2obs_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_namnest_precip_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_namnest_radar_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_namnest_severe_stats.sh
dev/drivers/scripts/cam/stats/jevs_cam_namnest_snowfall_stats.sh
dev/drivers/scripts/cam/plots/jevs_cam_grid2obs_plots.sh
dev/drivers/scripts/cam/plots/jevs_cam_headline_plots.sh
dev/drivers/scripts/cam/plots/jevs_cam_precip_plots.sh
dev/drivers/scripts/cam/plots/jevs_cam_snowfall_plots.sh
```
[Total: 13 prep jobs, 25 stats jobs, 4 plots jobs]

All jobs will be examined for successful completion.

- [x] Has the code been checked to ensure that no errors occur during the execution? **[No]**

The following was tested and completed successfully after the runtime bugfix was made:
jevs_cam_grid2obs_plots

The following selection were tested and completed successfully after EE2 edits:
jevs_cam_radar_prep.sh
jevs_cam_\*\_precip_prep.sh
jevs_cam_\*\_precip_stats.sh
jevs_cam_\*\_snowfall_stats.sh
jevs_cam_\*\_grid2obs_stats.sh
[all plots jobs]

- [x] Do these updates/additions include sufficient testing updates? **[Yes]**

- [x] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
I have cloned your fork and checked out branch `bugfix/CAM_EE2_fixes_and_plots_runtime_fix`.

File: scripts/cam/plots/exevs_cam_grid2obs_plots.sh [hunk 1]
```diff
@@ -70,8 +61,7 @@ if [ $USE_CFP = YES ]; then
         export MP_PGMMODEL=mpmd
         export MP_CMDFILE=${poe_script}
         if [ $machine = WCOSS2 ]; then
-            export LD_LIBRARY_PATH=/apps/dev/pmi-fix:$LD_LIBRARY_PATH
-            launcher="mpiexec -np $nproc -ppn $nproc --cpu-bind verbose,depth cfp"
+            launcher="mpiexec -np $nproc --cpu-bind verbose,depth cfp"
```
Comment (AliciaBentley-NOAA) [inline scripts/cam/plots/exevs_cam_grid2obs_plots.sh:64]: @MarcelCaron-NOAA Just wanted to check with you to see if this change to your launcher line was intentional. Mallory said that her code ran fine just by removing the /apps/dev/pmi-fix line above it. No other changes were needed.

File: jobs/JEVS_CAM_PREP [hunk 2]
```diff
@@ -5,6 +5,8 @@
 ########################################
 set -xa
 
+export SENDMAIL=${SENDMAIL:-NO}
```
Comment (AliciaBentley-NOAA) [inline jobs/JEVS_CAM_PREP:8]: @MarcelCaron-NOAA Can the full list of "export SEND____=" options be included in all j-jobs for NCO?

> export SENDMAIL=${SENDMAIL:-NO}
> export SENDCOM=${SENDCOM:-YES}
> export SENDDBN=${SENDDBN:-YES}
> export SENDECF=${SENDECF:-YES}

File: scripts/cam/plots/exevs_cam_grid2obs_plots.sh (hunk 1 above)
Comment (AliciaBentley-NOAA) [reply scripts/cam/plots/exevs_cam_grid2obs_plots.sh:64]: It also looks like the only script where you deleted _export LD_LIBRARY_PATH=/apps/dev/pmi-fix:$LD_LIBRARY_PATH_ and then edited the launcher line, and this deviation from other script edits seems odd enough to mention!

Comment (MarcelCaron-NOAA) [reply scripts/cam/plots/exevs_cam_grid2obs_plots.sh:64]: Thanks @AliciaBentley-NOAA for checking on these edits.  They are very minor changes but intentional.  Removing LD_LIBRARY_PATH is related to the EE2 fix, and removing the -ppn option for mpiexec is related to fixing the CAM grid2obs plots runtime/resource allocation.  I found that including this -ppn option resulted in overallocation warnings/errors, removing ensured the default is used.  This second change was only needed to fix this particular job and no others.

File: jobs/JEVS_CAM_PREP (hunk 2 above)
Comment (MarcelCaron-NOAA) [reply jobs/JEVS_CAM_PREP:8]: Sure, let me add that in now

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA Please also remove hardcoded paths to MET and METplus in /dev/drivers/scripts:

> export MET_PLUS_PATH="/apps/ops/prod/libs/intel/${intel_ver}/metplus/${metplus_ver}"
> export MET_PATH="/apps/ops/prod/libs/intel/${intel_ver}/met/${met_ver}"
> export MET_CONFIG="${MET_PLUS_PATH}/parm/met_config"

**Comment by MarcelCaron-NOAA** [comment]
Removed!

**Comment by MarcelCaron-NOAA** [comment]
MET_CONFIG was replaced with {ENV[METPLUS_PATH]}/parm/met_config and otherwise removed from parm/ and ush/ scripts.  I ran a test prep job to confirm the changes and the job completed successfully.

**Comment by ShelleyMelchior-NOAA** [comment]
The prep processing has been submitted and has completed.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/cam`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_*.o*` (latest timestamps)

File: dev/drivers/scripts/cam/stats/jevs_cam_hireswarw_grid2obs_stats.sh [hunk 3]
```diff
@@ -45,18 +45,16 @@ source /usr/share/lmod/lmod/init/sh
...
-export MET_PLUS_PATH="/apps/ops/prod/libs/intel/${intel_ver}/metplus/${metplus_ver}"
-export MET_PATH="/apps/ops/prod/libs/intel/${intel_ver}/met/${met_ver}"
-export MET_CONFIG="${MET_PLUS_PATH}/parm/met_config"
 export PYTHONPATH=$HOMEevs/ush/$COMPONENT:$PYTHONPATH
 
 # Developer Settings
 export envir=prod
 export DATAROOT=/lfs/h2/emc/stmp/$USER/evs_test/$envir/tmp
-export COMROOT=/lfs/h2/emc/vpppg/noscrub/${USER}
+export COMROOT=/lfs/h2/emc/vpppg/noscrub/emc.vpppg
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/cam/stats/jevs_cam_hireswarw_grid2obs_stats.sh]: @MarcelCaron-NOAA this one will need to revert to ${USER}

File: dev/drivers/scripts/cam/stats/jevs_cam_hireswarwmem2_precip_stats.sh [hunk 4]
```diff
@@ -59,4 +56,4 @@ export cyc=$(date -d "today" +"%H")
 export maillist="alicia.bentley@noaa.gov,marcel.caron@noaa.gov"
 
 # Job Settings and Run
-. ${HOMEevs}/jobs/cam/stats/JEVS_CAM_STATS
+. ${HOMEevs}/JEVS_CAM_STATS
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/cam/stats/jevs_cam_hireswarwmem2_precip_stats.sh]: Please add jobs/ subdir to this call to run JEVS_CAM_STATS. Should be:
```
. ${HOMEevs}/jobs/JEVS_CAM_STATS
```

**Comment by ShelleyMelchior-NOAA** [comment]
Starting to work through some stats processing.  First up, hireswarw and hireswarwmem2
### GOOD RUNS
jevs_cam_hireswarw_grid2obs_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230929/hireswarw/grid2obs/`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_grid2obs_stats.o83186604`

jevs_cam_hireswarw_radar_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230930/hireswarw/radar`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_radar_s
tats.o83217906`

jevs_cam_hireswarw_precip_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230926/hireswarw/precip`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_precip_stats.o83217535`

jevs_cam_hireswarw_snowfall_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230929/hireswarw/snowfall`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_snowfall_stats.o83217573`

jevs_cam_hireswarwmem2_grid2obs_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230929/hireswarwmem2/grid2obs/`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarwmem2_grid2obs_stats.o83186663`

jevs_cam_hireswarwmem2_precip_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230926/hireswarwmem2/precip`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarwmem2_precip_stats.o83216205`

jevs_cam_hireswarwmem2_snowfall_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230929/hireswarwmem2/snowfall`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarwmem2_snowfall_stats.o83217480`

### MISSING DATA
jevs_cam_hireswarw_severe_stats.sh 
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarw_severe_stats_o83218031`

jevs_cam_hireswarwmem2_severe_stats.sh
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarwmem2_severe_stats_00.o83216701`

jevs_cam_hireswarwmem2_radar_stats.sh
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_hireswarwmem2_radar_stats.o83186742`

**Comment by MarcelCaron-NOAA** [comment]
Hi @ShelleyMelchior-NOAA:

**Thanks very much for running the cam jobs this weekend and for the summary.  I can confirm that the following prep jobs completed normally (those that produced missing data messages are indicated with \*):**

jevs\_cam\_hireswarw\_precip\_prep
jevs\_cam\_hireswarwmem2\_precip\_prep
jevs\_cam\_hireswarw\_severe\_prep
jevs\_cam\_hireswarwmem2\_severe\_prep
jevs\_cam\_hireswfv3\_precip\_prep
jevs\_cam\_hireswfv3\_severe\_prep
jevs\_cam\_hrrr\_precip\_prep
jevs\_cam\_hrrr\_severe\_prep
jevs\_cam\_namnest\_precip\_prep
jevs\_cam\_namnest\_severe\_prep
_jevs\_cam\_radar\_prep*
jevs\_cam\_severe\_prep*_

**The following stats jobs completed cleanly, without unusual warnings, although the three you listed failed to process fully due to missing data (indicated with \*):**
jevs\_cam\_hireswarw\_grid2obs\_stats
jevs\_cam\_hireswarw\_precip\_stats
jevs\_cam\_hireswarw\_radar\_stats
_jevs\_cam\_hireswarw\_severe\_stats*_
jevs\_cam\_hireswarw\_snowfall\_stats
jevs\_cam\_hireswarwmem2\_grid2obs\_stats
jevs\_cam\_hireswarwmem2\_precip\_stats
_jevs\_cam\_hireswarwmem2\_radar\_stats*
jevs\_cam\_hireswarwmem2\_severe\_stats*_
jevs\_cam\_hireswarwmem2\_snowfall\_stats

**The following job failed:**
jevs_cam_href_severe_prep

This last job is within Binbin's area of responsibility.  I reverted the change made to that file in this PR and will allow that job to be covered in Binbin's PR!

**Comment by MarcelCaron-NOAA** [comment]
Hi @PerryShafran-NOAA the radar and severe prep and stats jobs italicized above (that produced missing data email messages) are those that I believe typically produce those messages, but I also wonder if it could be a timing issue.   Do you happen to know what cycles Logan has chosen to run these jobs in the parallel?

**Comment by MarcelCaron-NOAA** [comment]
Also, a quick question: the change in the location of the CAM J-job directory will break some jobs in the EVS parallel temporarily, until @PerryShafran-NOAA's and @BinbinZhou-NOAA's PRs are merged.  Instead of only changing the J-job paths in the CAM det driver scripts (excluding nam firewxnest), is it preferred that I change the paths in all CAM driver scripts in this PR?  @ShelleyMelchior-NOAA @AliciaBentley-NOAA

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA I hope that you saw my email this morning where I asked Perry and Binbin to get their CAM EE2 PRs in as soon as possible so that the CAM component of EVS is broken for the shortest amount of time after your CAM PR is merged!

@ShelleyMelchior-NOAA, @PerryShafran-NOAA, and @MarcelCaron-NOAA : Is something fundamentally wrong with the radar/severe scripts? They never seem to work for us. We've already removed the radar/severe plotting scripts from ops EVSv1. Should we remove prep/stats too, or can we figure this out so at least the prep files are done operationally?

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA Please remove the hardcoded path to /dcom on L79 of [jobs/cam/plots/JEVS_CAM_PLOTS](https://github.com/NOAA-EMC/EVS/blob/5210ee0c2b18b02e51df48b91d0e9a6a11f35f0d/jobs/cam/plots/JEVS_CAM_PLOTS#L79) 

(I think $DCOMROOT is a variable if you are loading prod utils, if this variable is even needed for CAM plotting scripts!)

**Comment by PerryShafran-NOAA** [comment]
@MarcelCaron-NOAA @ShelleyMelchior-NOAA @AliciaBentley-NOAA @BinbinZhou-NOAA Before we merge, we will need to resolve this discrepancy here in the JEVS_CAM_STATS file:

```
# LCD COMMENT: COMOUT should not be different between HREF and other CAMs
# I'm leaving this discrepancy for now, but we'll need to clean this up before EE2 review
if [ $MODELNAME = href ] ; then
   export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})}/$STEP/$COMPONENT
else
   export COMOUT=${COMOUT:-$(compath.py -o $NET/$evs_ver/$STEP/$COMPONENT/$RUN)}
fi
```

This should not be.  We need to have COMOUT be consistent across EVS, and it needs to end at $evs_ver, from what I understand.  This should be:

export COMOUT=${COMOUT:-$(compath.py ${envir}/com/${NET}/${evs_ver})}

Can we make sure this is the case across EVS (and especially across the CAM component)?  This will raise some red flags I feel with EE2 and with NCO.  This will require changes in the driver script and maybe people's exevs scripts.  

Perry

**Comment by MarcelCaron-NOAA** [comment]
> @MarcelCaron-NOAA I hope that you saw my email this morning where I asked Perry and Binbin to get their CAM EE2 PRs in as soon as possible so that the CAM component of EVS is broken for the shortest amount of time after your CAM PR is merged!

Yes, I did.  Thanks!

> @ShelleyMelchior-NOAA, @PerryShafran-NOAA, and @MarcelCaron-NOAA : Is something fundamentally wrong with the radar/severe scripts? They never seem to work for us. We've already removed the radar/severe plotting scripts from ops EVSv1. Should we remove prep/stats too, or can we figure this out so at least the prep files are done operationally?

It does look like relatively recent jobs have completed at least partially successfully in the parallel.  See the results of the following:
```
du -sh /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/*/*severe*stat
du -sh /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/*/*radar*stat
```

[Note: The severe stats job uses PDYm7, so the most recent stats are from 9/25]

**Comment by ShelleyMelchior-NOAA** [comment]
hrrr stats processing has been submitted.

jevs_cam_hrrr_precip_stats.sh:
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230926/hrrr/precip/`

jevs_cam_hrrr_snowfall_stats.sh:
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230930/hrrr/snowfall`

jevs_cam_hrrr_radar_stats.sh:
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/hrrr/radar`

jevs_cam_hrrr_severe_stats.sh:
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/hrrr.20230925`

UPDATE: hrrr_grid2obs_stats has completed.
jevs_cam_hrrr_grid2obs_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/hrrr/grid2obs`

File: dev/drivers/scripts/cam/stats/jevs_cam_namnest_severe_stats.sh
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/cam/stats/jevs_cam_namnest_severe_stats.sh:1]: Please adjust walltime to be: `#PBS -l walltime=00:30:00`

File: dev/drivers/scripts/cam/stats/jevs_cam_namnest_radar_stats.sh
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/cam/stats/jevs_cam_namnest_radar_stats.sh:1]: Please adjust walltime to be: `#PBS -l walltime=00:30:00`

**Comment by ShelleyMelchior-NOAA** [comment]
namnest stats processing has been submitted.

jevs_cam_namnest_precip_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230926/namnest/precip/`

jevs_cam_namnest_radar_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/namnest/radar`

jevs_cam_namnest_severe_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/namnest.20230925`

jevs_cam_namnest_snowfall_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230930/namnest/snowfall`

UPDATE: namnest_grid2obs_stats completed
jevs_cam_namnest_grid2obs_stats.sh
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/namnest/grid2obs`

File: jobs/JEVS_CAM_STATS [hunk 5]
```diff
@@ -5,6 +5,7 @@
 ########################################
 set -xa
 
+export SENDMAIL=${SENDMAIL:-NO}
 export SENDCOM=${SENDCOM:-YES}
 export SENDDBN=${SENDDBN:-YES}
```
Comment (AliciaBentley-NOAA) [inline jobs/JEVS_CAM_STATS]: @MarcelCaron-NOAA Please use _export SENDDBN=${SENDDBN:-**NO**}_ in your CAM j-jobs. 

@PerryShafran-NOAA and @BinbinZhou-NOAA CCed for awareness to a J-job change.

**Comment by ShelleyMelchior-NOAA** [comment]
plots processing is underway. I will update as processes complete.
```
       Req'd  Req'd   Elap
Job ID                         Username        Queue           Jobname         SessID   NDS  TSK   Memory Time  S Time
------------------------------ --------------- --------------- --------------- -------- ---- ----- ------ ----- - -----
83338646.cbqs01                shelley.melchi* dev             jevs_cam_preci*     6895    4   512    --  05:30 R 00:04
83338672.cbqs01                shelley.melchi* dev             jevs_cam_snowf*    93807    4   512    --  05:30 R 00:02
83338696.cbqs01                shelley.melchi* dev             jevs_cam_headl*    95329    1   128    --  00:30 R 00:02
83338718.cbqs01                shelley.melchi* dev             jevs_cam_grid2*   158121    4   512    --  10:00 R 00:01
```

**Comment by MarcelCaron-NOAA** [comment]
> hrrr stats processing has been submitted.
> 
> jevs_cam_hrrr_precip_stats.sh: COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230926/hrrr/precip/`
> 
> jevs_cam_hrrr_snowfall_stats.sh: COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230930/hrrr/snowfall`
> 
> jevs_cam_hrrr_radar_stats.sh: COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/hrrr/radar`
> 
> jevs_cam_hrrr_severe_stats.sh: COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/hrrr.20230925`
> 
> UPDATE: hrrr_grid2obs_stats has completed. jevs_cam_hrrr_grid2obs_stats.sh COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/hrrr/grid2obs`

All of the five HRRR jobs completed cleanly and the output are as expected.

File: jobs/JEVS_CAM_STATS (hunk 5 above)
Comment (MarcelCaron-NOAA) [reply jobs/JEVS_CAM_STATS]: Done!

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA Thanks! Please also do a quick search for "set -xa" in your J-jobs and change it to "set -x" (without the "a"). Thanks!

**Comment by ShelleyMelchior-NOAA** [comment]
Headline plots has completed:
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/cam/atmos.20231001`

**Comment by MarcelCaron-NOAA** [comment]
> namnest stats processing has been submitted.
> 
> jevs_cam_namnest_precip_stats.sh COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230926/namnest/precip/`
> 
> jevs_cam_namnest_radar_stats.sh COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/namnest/radar`
> 
> jevs_cam_namnest_severe_stats.sh COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/namnest.20230925`
> 
> jevs_cam_namnest_snowfall_stats.sh COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20230930/namnest/snowfall`
> 
> UPDATE: namnest_grid2obs_stats completed jevs_cam_namnest_grid2obs_stats.sh COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231001/namnest/grid2obs`

All of the five NAM Nest stats jobs have completed cleanly and the output look normal.

**Comment by ShelleyMelchior-NOAA** [comment]
Snowfall plots has completed.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/cam/atmos.20231001`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_snowfall_plots.o83338672`

**Comment by ShelleyMelchior-NOAA** [comment]
precip plots has completed.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/cam/atmos.20231001`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_precip_plots.o83338646`

**Comment by ShelleyMelchior-NOAA** [comment]
grid2obs plots has completed.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/cam/atmos.20231001`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_grid2obs_plots.o83338718`

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA Binbin made the following change in a CAM plots J-job. Can you please implement this change in your CAM J-jobs as well if it won't break anything of yours?

  export COMINapcp24mean=${COMINspatial:-$(compath.py  $envir/com/$NET/$evs_ver)}/stats/$COMPONENT
  changes to
  export COMINapcp24mean=${**COMINapcp24mean**:-$(compath.py  $envir/com/$NET/$evs_ver)}/stats/$COMPONENT
  
He also changed the following in dev/drivers/scripts/cam/plots/jevs_cam_href_precip_spatial_plots.sh
export COMINspatial=/lfs/h2/emc/vpppg/noscrub/${USER}/$NET/$evs_ver
  changes to
export **COMINapcp24mean**=/lfs/h2/emc/vpppg/noscrub/${USER}/$NET/$evs_ver

**Comment by MarcelCaron-NOAA** [comment]
@AliciaBentley-NOAA @BinbinZhou-NOAA Got it!  I don't think these will interfere with any CAM det jobs already tested in this PR.

**Comment by MarcelCaron-NOAA** [comment]
@ShelleyMelchior-NOAA All four plots jobs completed cleanly and all of the output looks normal.  Notably, the CAM grid2obs plots job took about 6h30m to complete, as intended, no killed job cards.

Let me know if I missed anything.  Otherwise this looks good to me!

**Comment by ShelleyMelchior-NOAA** [comment]
Yes, @MarcelCaron-NOAA , I noted the quicker completions as well. Awesome! I will do the formal code review, though I have been doing it as I go along. I will coordinate w/ @PerryShafran-NOAA for when to merge this PR, since we are trying to merge the cam components generally at the same time. But it will be soon!

**Comment by AliciaBentley-NOAA** [comment]
Hi @MarcelCaron-NOAA. Since this PR isn't merged yet, I'd like to propose some quick SENDDBN changes (that shouldn't require any more testing) that will make this pristine for NCO.

1) Please set "export SENDDBN=${SENDDBN:-NO}" in your prep and stats J-jobs (I think they are already)!
2) Please use "export SENDDBN=${SENDDBN:-YES}" in your plots J-jobs. NCO will want SENDDBN set to YES in production eventually.
3) We don't actually want to send NCO any SENDDBN alerts yet, please add "export SENDDBN=NO" into your plotting dev driver scripts.

Thank you!

**Comment by MarcelCaron-NOAA** [comment]
Okay @AliciaBentley-NOAA, I checked these scripts and changed the default setting for SENDDBN in the plots J-job.  Everything else should be as you proposed.

**Comment by ShelleyMelchior-NOAA** [comment]
I am going to begin the merging process for this PR. That will get these changes into develop and allow for the other 2 cam PRs to adopt the J-job scripts to all be the same across all of cam's sub-components. Coordinating w/ @PerryShafran-NOAA , he will not put these develop updates into para until all cam PRs are merged.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
Testing produced expected results. This PR is ready for merge.

**Pull request merged**